db/machine.go

```diff
@@ -49,25 +50,265 @@ func initMachines(tx *sql.Tx) error {
+
+		m := Machine{db: &d}
+		err = rows.Scan(
+			&m.Id, &m.Name, &m.Architecture, &m.Microarchitecture,
+			&m.Cores, &m.MemoryGB)
+		if err != nil {
+			return nil, err
+		}
+
+		sortmemory = append(sortmemory, m)
+	}
+
+	return sortmemory, rows.Err()
+
+}
+
+//Sort by mirchroarch
+func (d DB) Sort_Microarch(microarch1 string)([]Machine, error){
+	rows, err := d.sql.Query(`
+		SELECT id, name, arch, microarch, cores, memory
+		FROM Machines
+		WHERE microarch = ?
+		ORDER BY memory DESC`, microarch1)
+	if err != nil {
+		return nil, err
+	}
+
+	defer rows.Close()
+
+	sortmicroarch := []Machine{}
+	for rows.Next() {
+
+		m := Machine{db: &d}
+		err = rows.Scan(
+			&m.Id, &m.Name, &m.Architecture, &m.Microarchitecture,
+			&m.Cores, &m.MemoryGB)
+		if err != nil {
+			return nil, err
+		}
+
+		sortmicroarch = append(sortmicroarch, m)
+	}
+
+	return sortmicroarch, rows.Err()
+
+}
+//Sort machine by architecture
+func (d DB) Sort_By_Arch(arch string)([]Machine, error){
+	rows, err := d.sql.Query(`
+		SELECT id, name, arch, microarch, cores, memory
+		FROM Machines
+		WHERE arch = ?
+		ORDER BY arch DESC`, arch)
+	if err != nil {
+		return nil, err
+	}
+
+	defer rows.Close()
+
+	sortarch := []Machine{}
+	for rows.Next() {
+
+		m := Machine{db: &d}
+		err = rows.Scan(
+			&m.Id, &m.Name, &m.Architecture, &m.Microarchitecture,
+			&m.Cores, &m.MemoryGB)
+		if err != nil {
+			return nil, err
+		}
+
+		sortarch = append(sortarch, m)
+	}
+
+	return sortarch, rows.Err()
+
+}
+//Sort machine by name
+func (d DB) Sort_M_By_Name()([]Machine, error){
+	rows, err := d.sql.Query(`
+		SELECT id,name, arch, microarch, cores, memory
+		FROM Machines
+		ORDER BY LOWER (name) COLLATE NOCASE ASC	
+	`)
+	if err != nil {
+		return nil, err
+	}
+
+	defer rows.Close()
+
+	sort_name := []Machine{}
+	for rows.Next() {
+		m := Machine{db: &d}
+		err = rows.Scan(
+			&m.Id, &m.Name, &m.Architecture, &m.Microarchitecture,
+			&m.Cores, &m.MemoryGB)
+
+		if err != nil {
+			return nil, err
+		}
+
+		sort_name = append(sort_name, m)
+	}
+
+	return sort_name, rows.Err()
+
+}
+
+
+//FILTER BY MEMORY SIZE
+func (d DB) Filter_By_Memory(min int, max int)([]Machine, error){
+	rows, err := d.sql.Query(`
+		SELECT id, name, arch, microarch, cores, memory
+		FROM Machines
+		WHERE memory >= ? AND memory <= ?
+		`, min, max)
+	if err != nil {
+		return nil, err
+	}
+
+	defer rows.Close()
+
+	filtermemory := []Machine{}
+	for rows.Next() {
+
+		m := Machine{db: &d}
+		err = rows.Scan(
+			&m.Id, &m.Name, &m.Architecture, &m.Microarchitecture,
+			&m.Cores, &m.MemoryGB)
+		if err != nil {
+			return nil, err
+		}
+
+		filtermemory = append(filtermemory, m)
+	}
+
+	return filtermemory, rows.Err()
+
+}
 
+
+//GET MACHINE: DISPLAY A MACHINE DATAILS
 func (d DB) GetMachine(column string, val interface{}) (*Machine, error) {
-	row := d.sql.QueryRow(`
+	rows := d.sql.QueryRow(`
 		SELECT id, name, arch, microarch, cores, memory
 		FROM Machines
-		WHERE Machines.`+column+` = $1`, val)
```

Comment:
Why was it necessary to add additional qualification here? If there is only one table involved in the query, there shouldn't be any name collisions, right?